web-ng/src/app/services/router/router.service.ts

```diff
@@ -184,26 +168,31 @@ export class NavigationService {
    * Navigate to the current URL, updating the observation id in the URL
    * fragment.
    */
-  setObservationId(id: string | null) {
-    this.setFragmentParam(
-      Map([[NavigationService.OBSERVATION_ID_FRAGMENT_PARAM, id]])
-    );
+  editObservation(featureId: string | null, observationId: string) {
+    const newParam: { [key: string]: string } = {};
+    if (featureId) {
+      newParam[NavigationService.FEATURE_ID_FRAGMENT_PARAM] = featureId!;
+    }
+    newParam[NavigationService.OBSERVATION_ID_FRAGMENT_PARAM] = observationId;
+    this.setFragmentParams(new HttpParams({ fromObject: newParam }));
   }
 
-  editObservation(id: string | null) {
-    this.setFragmentParam(
-      Map([[NavigationService.OBSERVATION_ID_FRAGMENT_PARAM, id]])
-    );
+  clearObservationId(featureId: string | null) {
```

Comment:
Is `featureId` necessary here? Based on its name I'd assume this method would update the current URL, removing the observation id. Also, it should probably be called something like `clearObservation` for consistency with other methods.